### PR TITLE
105551: Add batch observations API endpoint

### PR DESCRIPTION
**Pull request by vvkpd**
## Summary
Add REST endpoint for fetching observations in batch across multiple visits, enabling efficient data retrieval for care view components.

## Changes
- **BahmniObservationsBatchRequest.java**: New request contract supporting batch observation queries with optional filtering (scope, obsIgnoreList, filterObsWithOrders)
- **VisitObservationsResponse.java**: New response contract mapping visit UUID to observations
- **BahmniObservationsController.java**: Added POST /observations/batch endpoint accepting batch requests and returning visit-observation mappings
- **BahmniObservationsControllerTest.java**: Test coverage for batch endpoint with various filter scenarios

## Related Work
Supports frontend feature: displaying new care instructions in IPD care view (openmrs-module-ipd-frontend#46) by providing efficient backend batch API for multi-visit observation fetching.

## Test Coverage
- Unit tests for batch request/response contracts
- Controller integration tests for batch endpoint with filters
- Edge cases: empty visits list, missing concepts, various filter combinations

## Note
This work complements the frontend implementation in openmrs-module-ipd-frontend#46 which fetches care instructions for multiple patients in a single batch call.

### CODE REVIEW

File: bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/display/controls/BahmniObservationsController.java [hunk 1]
```diff
@@ -82,6 +85,34 @@ public Collection<BahmniObservation> get(@RequestParam(value = "patientUuid", re
         return observations;
     }
 
+    @RequestMapping(value = "/batch", method = RequestMethod.POST)
+    @ResponseBody
+    public List<VisitObservationsResponse> getBatch(@RequestBody BahmniObservationsBatchRequest request) {
```
Comment (binduak) [inline bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/display/controls/BahmniObservationsController.java:90]: @vvkpd if i understand correctly it gets the obs for the mentioned concepts within one visit ? can you please confirm
also why this has be POST method ?

Comment (vvkpd) [reply bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/display/controls/BahmniObservationsController.java:90]: Yes, it is the extended version of /observations?visitUuid=visit-uuid-1&concept=Physician%20Orders%20Comments&concept=Instruction%20for%20the%20Ward

Please refer the ADR doc here: https://github.com/cureinternational/bahmni-core/blob/nurse-acknowledge/docs/ADR_105551_batch_observations_api.md

**Comment by vvkpd** [comment]
The current implementation demonstrates acceptable and stable performance characteristics for batch requests. Based on testing:

~10 visit UUIDs → ~2 seconds
~40+ visit UUIDs → ~3 seconds
